src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509ChainProcessor.cs

```diff
@@ -30,13 +32,16 @@ internal sealed class OpenSslX509ChainProcessor : IChainPal
         private static readonly CachedDirectoryStoreProvider s_userPersonalStore =
             new CachedDirectoryStoreProvider(X509Store.MyStoreName);
 
+        private static readonly ConcurrentDictionary<Interop.Crypto.X509VerifyStatusCode, string> s_errorStrings =
+            new ConcurrentDictionary<Interop.Crypto.X509VerifyStatusCode, string>();
```

Comment:
Can you add a comment about why it's worth having this cache for error strings?  Presumably it's because we retrieve these in cases where they're not actually considered errors?

Reply:
Yeah, particularly if a chain is built with suppression flags, then things like "unable to get certificate CRL" will get copied over and over again.  In the Windows version we preload all of the strings (we'll ever care about) during static field init.